src/app/[locale]/dashboard/products/_components/all-products-table.tsx

```
@@ -0,0 +1,125 @@
+    startTransition(async () => {
+        try {
+        const result = await deleteMutation.mutateAsync(id);
+        if (!result.ok) {
+            toast.error(result.error || t("failed-delete"));
+            return;
+        }
+        toast.success(t("success-delete"));
+
+        // revalidate data after delete
+        if (onDelete) onDelete();
+        } catch (err: any) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)